File: java/ankit_kanyal/devnest_problems/DeckOfCards.java
// link- https://devsnest.in/algo-challenges/deck-of-cards

static int solve(int n, int[] deck){
//CODE HERE 
    int len = deck.length;
    if(len == 1) return 0;
        
    HashMap<Integer, Integer> map = new HashMap<Integer, Integer>();
    for(int i:deck) {
        if(map.containsKey(i)) map.put(i, map.get(i)+1);
        else map.put(i,1);
    }
        
        
    int min = map.get(deck[0]);        
    for(int i:map.values()) min = gcd(min,i);
    if(min!=1) return 1;
    return 0;
    
}

 public static int gcd(int n1,int n2){
        if(n2!=0) return gcd(n2,n1%n2);
        return n1;
    }

File: python/vedant_bhatnagar/DevsnestProblems_dailyUpdated/DeckOfCards.py
def gcd(a,b):
    if b==0:
        return a
    return gcd(b, a%b)

def solve(n, deck):
    # CODE HERE
    m = {}
    if n<2:
        return 0
    for i in deck:
        if i in m:
            m[i] += 1
        else:
            m[i] = 1
    
    track = list(m.values())
    curr_gcd = track[0]
    for i in track:
        if curr_gcd>i:
            curr_gcd = gcd(curr_gcd, i)
        else:
            curr_gcd = gcd(i,curr_gcd)
    if curr_gcd > 1:
        return 1
    else:
        return 0